Update Documentation: create an upgrade rule, snippet rule, and update the README in docs for AI (#37459)
gradle/gradle: platforms/documentation/docs/src/docs/skills/upgrade-guide/SKILL.md

---

> name: upgrade-guide

Reviewer: The name `upgrade-guide` is a bit vague. What about calling it `gradle-upgrade`? It makes it clear that it targets Gradle.

Reviewer 2: Or `gradle-upgrade-assistant` even, since the skill does not actually upgrade a build.

> name: upgrade-guide

Reviewer 2: Should all of this say "update" instead of "upgrade"?

> description: Gradle upgrade guide covering versions 8.0 through 9.5.0. Use when helping users upgrade Gradle, diagnose version-related build failures, or understand breaking changes between versions.

Reviewer 2: ```suggestion
description: Assists with upgrading Gradle from versions 8.0 through 9.5.0. Use to help users upgrade, diagnose upgrade-related build failures, or understand breaking changes between versions.
```

argument-hint: [from-version] [to-version]
---

# Gradle Upgrade Guide

> Help the user upgrade their Gradle build between versions, or answer questions about breaking changes, deprecations, and migration strategies.

Reviewer 2: ```suggestion
Help upgrade the Gradle version used in a build by answering questions about breaking changes, deprecations, and migration strategies.
```
